### DISCUSSION TITLE
Meaning of groupby_bins.groups

### DISCUSSION THREAD

**Discussion by mwasserstein** [pydata/xarray#6830] [Q&A]
Hello:

I am attempting to use groupby_bins to bin some data in equal size bins. To do so I specify an integer number of bins that I want the data to be binned in. However I am skeptical that the binning is being done correctly, based on the results that I get. To investigate, I found the `.groups` feature, and I'm wondering exactly what this means.  Here is a photo of my code and the output.
<img width="1026" alt="Screen Shot 2022-07-26 at 11 37 59 AM" src="https://user-images.githubusercontent.com/71135396/181074110-4a8f7e98-e33e-4924-bafc-2887e89af066.png">

I'm specifically wondering about the integers that get listed after each datetime, which represents the label of my bins.

Thanks for the help.

**Comment by dcherian** [comment]
Those are the indices along `time` that belong to a specific group or bin.

**Reply by mwasserstein** [reply]
Ok, so if that is the case, it appears that my bins are not of equal size. For instance, the second bin is larger than the first, which is larger than the third. Why could this be? I specify in my code that I want an integer number of equal size bins, using the `bins=` argument.

**Reply by dcherian** [reply]
I would investigate what [pandas.cut](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.cut.html) does. That's what we use to create the groups.
